packages/api/src/users/commands/handlers/registerUser.handler.ts

```diff
@@ -0,0 +1,26 @@
+import { BadRequestException } from '@nestjs/common';
+import { CommandHandler, EventBus, ICommandHandler } from '@nestjs/cqrs';
+
+import { UserErrorMessage } from '@coderscamp/shared/api';
+import type { RegisteredUser } from '@coderscamp/shared/models';
+
+import { UserRegisteredEvent } from '../../events';
+import { UsersRepository } from '../../users.repository';
+import { RegisterUserCommand } from '../registerUser.command';
+
+@CommandHandler(RegisterUserCommand)
+export class RegisterUserHandler implements ICommandHandler<RegisterUserCommand> {
+  constructor(private readonly eventBus: EventBus, private readonly repository: UsersRepository) {}
+
+  async execute({ input: userData }: RegisterUserCommand): Promise<RegisteredUser> {
+    const userFromDataBase = await this.repository.findByGithubId(userData.githubId);
+
+    if (userFromDataBase) throw new BadRequestException(UserErrorMessage.USER_ALREADY_EXISTS);
+
+    const user = await this.repository.create(userData);
+
+    this.eventBus.publish(new UserRegisteredEvent(user));
```

Comment:
I'm in favor of names like: UserWasRegistered.
More about naming here: 

https://verraes.net/2019/06/messaging-patterns-natural-language-message-names/